python/pyspark/sql/types.py

```diff
@@ -653,8 +653,8 @@ def fromJson(cls, json: Dict[str, Any]) -> "StructField":
         return StructField(
             json["name"],
             _parse_datatype_json_value(json["type"]),
-            json["nullable"],
-            json["metadata"],
+            json.get("nullable", True),
+            json.get("metadata"),
```

Comment:
When is this not set? This JSON ser/de is an internal format and not meant to be used by end users directly.

Reply:
Here an example, many dataframes are being created from a schema, this schema is created from a Json.
The input parameters to create a schema is StructType.fromJson(json), this internally uses StructField.fromJson().

The issue is when the StructField parses the Json, which forces to define the nullable and metadata attributes inside.
![image](https://user-images.githubusercontent.com/7476964/196637396-d437278c-f462-41dd-8323-3d613c05214b.png)

it is understandable that name and type are mandatory, but the others should be optional. 

The current parsing does not allow this. If more than 1000 fields are defined, this would be a headache and unnecessary entry.

Reply:
the way to avoid this issue users would be to declare the nullable and metadata attributes mandatory:
Example:
{
      "name": "auditname",
      "type": "string",
      "nullable": true, -> mandatory
      "metadata": {} -> mandatory
    }

I have avoided skipping this by not using the fromJson method, I mean passing the json directly.

json = {
      "name": "auditname",
      "type: StringType()
    }
StructField(**json)

this gets tricky because it is necessary to add some previous logics to do the above, the solution would be to use .get in the fromJson method. 

def fromJson(cls, json: Dict[str, Any]) -> "StructField":
        return StructField(
            json["name"],
            _parse_datatatype_json_value(json["type"]),
            json.get("nullable", True), -> Use get
            json.get("metadata"), -> Use get
        )